### PR TITLE
Add dartdocs to asset_bundle.dart

### LINKED ISSUE #3077: AssetBundle class needs documentation
The AssetBundle doc:

http://docs.flutter.io/flutter/services/AssetBundle-class.html

Does not mention you need to mark-up the flutter.yaml file to add the specific assets so they can be loaded later. Please add this so devs don't have to look for examples :-)

### PR DESCRIPTION
**Pull request by abarth**
Fixes #3077

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/services/asset_bundle.dart [hunk 1]
```diff
@@ -15,16 +15,51 @@ import 'image_decoder.dart';
...
+/// application's user interface.
+///
+/// Applications have a [rootBundle], which contains the resources that were
+/// packaged with the application when it was built. To add resources to the
+/// [rootBundle] for your application, add them to the `assets` section of your
+/// `flutter.yaml` manifest.
+///
+/// Rather than accessing the [rootBundle] global static directly, consider
+/// obtaining the [AssetBundle] for the current [BuildContext] using
+/// [DefaultAssetBundle.of]. This layer of indirection lets containing widget
```
Comment (Hixie) [inline packages/flutter/lib/src/services/asset_bundle.dart]: "widgets", though I'm not sure what "containing" means in this context

File: packages/flutter/lib/src/services/asset_bundle.dart [hunk 2]
```diff
@@ -77,9 +117,12 @@ abstract class CachingAssetBundle extends AssetBundle {
   }
 }
 
+/// An [AssetBundle] that loads resources from a Mojo service.
 class MojoAssetBundle extends CachingAssetBundle {
+  /// Creates an [AssetBundle] interface around the given [AssetBundleProxy] Mojo service.
   MojoAssetBundle(this._bundle);
 
+  /// Retreives the asset bundle located at the given URL, unpacks it, and provides it contents.
```
Comment (Hixie) [inline packages/flutter/lib/src/services/asset_bundle.dart]: retrieves

File: packages/flutter/lib/src/services/asset_bundle.dart [hunk 3]
```diff
@@ -48,19 +86,21 @@ class NetworkAssetBundle extends AssetBundle {
...
 
+/// An [AssetBundle] that adds a layer of caching to an asset bundle.
 abstract class CachingAssetBundle extends AssetBundle {
-  final Map<String, ImageResource> imageResourceCache =
+  final Map<String, ImageResource> _imageResourceCache =
     <String, ImageResource>{};
   final Map<String, Future<String>> _stringCache =
     <String, Future<String>>{};
 
+  /// Override to alter how images are retrieved from the underlying [AssetBundle].
```
Comment (Hixie) [inline packages/flutter/lib/src/services/asset_bundle.dart]: maybe mention that this is used by the resolution-aware bundle?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/services/asset_bundle.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/services/asset_bundle.dart]: Changed to ancestor.

**Pull request merged**